### PR TITLE
[Dashboard] Remove tw-components <Badge />

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on refactoring badge components across various files, simplifying their usage and styling. It removes unnecessary props and imports, standardizing the way `Badge` components are utilized throughout the application.

### Detailed summary
- Deleted `Badge` component from `apps/dashboard/src/tw-components/badge.tsx`.
- Simplified `OpenSeaPropertyBadge` by removing props and using a default `Badge`.
- Updated `BillingHeader`, `PlanCard`, and `BillingPlanCard` to use simpler `Badge` implementations.
- Removed `Badge` imports where not needed in multiple files.
- Adjusted `AccountSigner` and `ContractSourcesPage` to streamline `Badge` usage with fewer props.
- Changed button variant in `ContractSourcesPage` from `solid` to `primary`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5370?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5370** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5370?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5369** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5369?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5368** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5368?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5367** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5367?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/account-permissions/components/account-signer.tsx [hunk 1]
```diff
@@ -46,16 +47,12 @@ export const AccountSigner: React.FC<AccountSignerProps> = ({
...
               {isAdmin ? (
-                <Badge borderRadius="lg" p={1.5}>
-                  Admin Key
-                </Badge>
+                <Badge className="rounded-lg p-1.5">Admin Key</Badge>
               ) : (
-                <Badge borderRadius="lg" p={1.5}>
-                  Scoped key
-                </Badge>
+                <Badge className="rounded-lg p-1.5">Scoped key</Badge>
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/account-permissions/components/account-signer.tsx:52]: Remove the padding and rounded classes here - just use the default styling

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/sources/ContractSourcesPage.tsx [hunk 2]
```diff
@@ -91,7 +94,7 @@ const VerifyContractModal: React.FC<
         <ModalHeader>
           <Flex gap={2} align="center">
             <Heading size="subtitle.md">Contract Verification</Heading>
-            <Badge variant="outline" colorScheme="purple" rounded="md" px={2}>
+            <Badge variant="outline" className="px-2">
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/sources/ContractSourcesPage.tsx:97]: keep the default padding

File: apps/dashboard/src/components/badges/opensea.tsx [hunk 3]
```diff
@@ -1,17 +1,10 @@
...
     <ToolTipLabel label="This property is supported on OpenSea">
-      <Badge
-        {...props}
-        borderRadius="full"
-        variant="outline"
-        colorScheme="opensea"
-      >
-        OpenSea
-      </Badge>
+      <Badge className="rounded-full text-blue-500">OpenSea</Badge>
```
Comment (MananTank) [inline apps/dashboard/src/components/badges/opensea.tsx:7]: remove className - use the default variant

File: apps/dashboard/src/components/homepage/sections/PricingCard.tsx [hunk 4]
```diff
@@ -69,13 +69,7 @@ export const PricingCard: React.FC<PricingCardProps> = ({
...
             </Heading>
             {current && (
-              <Badge
-                borderRadius="md"
-                size="label.sm"
-                px={3}
-                py={1.5}
-                textTransform="capitalize"
-              >
+              <Badge className="rounded-md px-3 py-1.5 capitalize">
```
Comment (MananTank) [inline apps/dashboard/src/components/homepage/sections/PricingCard.tsx:72]: Remove className - use default variant

File: apps/dashboard/src/components/onboarding/ApplyForOpCreditsModal.tsx [hunk 5]
```diff
@@ -114,13 +115,7 @@ export const ApplyForOpCreditsModal: React.FC = () => {
...
               <Box position="absolute">
                 <Badge
-                  borderRadius="full"
-                  size="label.sm"
-                  px={3}
-                  bgColor={creditsRecord.color}
-                  color="#fff"
-                  textTransform="capitalize"
-                  fontWeight="bold"
+                  className={`font-bold text-white capitalize bg-[${creditsRecord.color}] rounded-full px-3`}
```
Comment (MananTank) [inline apps/dashboard/src/components/onboarding/ApplyForOpCreditsModal.tsx:118]: You can't do dynamic styles like this with tailwind - tailwind styles are generated at compile time - use `style` prop here for setting background

File: apps/dashboard/src/components/onboarding/PlanCard.tsx [hunk 6]
```diff
@@ -12,13 +13,7 @@ export const PlanCard: React.FC<PlanCardProps> = ({ creditsRecord }) => {
...
         <div>
           <Badge
-            borderRadius="full"
-            size="label.sm"
-            px={3}
-            bgColor={creditsRecord.color}
-            color="#fff"
-            textTransform="capitalize"
-            fontWeight="bold"
+            className={`font-bold text-white capitalize bg-[${creditsRecord.color}] rounded-full px-3`}
```
Comment (MananTank) [inline apps/dashboard/src/components/onboarding/PlanCard.tsx:16]: same thing here - use `style` prop for background

File: apps/dashboard/src/components/settings/Account/Billing/Header.tsx [hunk 7]
```diff
@@ -20,13 +21,7 @@ export const BillingHeader: React.FC<BillingHeaderProps> = ({
...
         </Text>
 
-        <Badge
-          borderRadius="md"
-          size="label.sm"
-          px={3}
-          py={1.5}
-          textTransform="capitalize"
-        >
+        <Badge className="rounded-md px-3 py-1.5 capitalize">
```
Comment (MananTank) [inline apps/dashboard/src/components/settings/Account/Billing/Header.tsx:24]: remove className - use outline variant here

File: apps/dashboard/src/components/settings/Account/Billing/PlanCard.tsx [hunk 8]
```diff
@@ -19,13 +20,7 @@ export const BillingPlanCard = () => {
...
       <div className="flex flex-row items-center gap-2">
         <Heading size="title.xs">Your current plan is</Heading>
-        <Badge
-          borderRadius="md"
-          size="label.sm"
-          textTransform="capitalize"
-          px={3}
-          py={1.5}
-        >
+        <Badge className="rounded-md px-3 py-1.5 capitalize">
```
Comment (MananTank) [inline apps/dashboard/src/components/settings/Account/Billing/PlanCard.tsx:23]: remove classname - use default variant

File: apps/dashboard/src/components/homepage/sections/PricingCard.tsx [hunk 9]
```diff
@@ -69,13 +69,7 @@ export const PricingCard: React.FC<PricingCardProps> = ({
...
                 Current plan
```
Comment (MananTank) [inline apps/dashboard/src/components/homepage/sections/PricingCard.tsx:73]: Fix the vertical alignment

**Review by MananTank — CHANGES_REQUESTED** [review]
please address changes

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/account-permissions/components/account-signer.tsx (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/account-permissions/components/account-signer.tsx:52]: thanks! I haven't tested this PR yet (the local dashboard thingy is not running - will check later today)

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/sources/ContractSourcesPage.tsx (hunk 2 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/sources/ContractSourcesPage.tsx:97]: Badge doesn't have primary variant tho, only `"outline" | "default" | "destructive" | "secondary" | "success" | "warning"`

Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/sources/ContractSourcesPage.tsx]: yeah - the default is primary

Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/sources/ContractSourcesPage.tsx]: updated!

**Review by MananTank — APPROVED** [review]

**Pull request merged**